.github/workflows/sonar-maven.yml

```diff
@@ -52,7 +56,7 @@ jobs:
           fi
           echo "sonar_args=$SONAR_ARGS" >> $GITHUB_OUTPUT
 
-          CHECK_REF=main
+          CHECK_REF=${{ inputs.commit_sha }}
```

Comment:
AFAIK this won't work without additional configuration to SonarScanner plugin that needs to know what branch it is working with.